pkg/processor/build/builder.go

```diff
@@ -743,15 +746,53 @@ func (b *Builder) validateAndParseS3Attributes(attributes map[string]interface{}
 
 func (b *Builder) getFunctionPathFromGithubURL(functionPath string) (string, error) {
 	if branch, ok := b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["branch"]; ok {
-		functionPath = fmt.Sprintf("%s/archive/%s.zip",
-			strings.TrimRight(functionPath, "/"),
-			branch)
+		var err error
+		functionPath, err = b.generateGithubZipballURL(functionPath, branch.(string))
+		if err != nil {
+			return "", errors.Wrap(err, "Failed to generate github zipball URL")
+		}
 	} else {
 		return "", errors.New("If code entry type is github, branch must be provided")
 	}
 	return functionPath, nil
 }
 
+func (b *Builder) generateGithubZipballURL(functionPath, branch string) (string, error) {
+	// split the URL to get the ORG and REPO using regex
+	org, repo, err := b.resolveGithubUrlComponents(functionPath)
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to resolve github URL components")
+	}
+
+	// create the URL for downloading the zipball Github api
+	return fmt.Sprintf("https://api.github.com/repos/%s/%s/zipball/%s", org, repo, branch), nil
+}
+
+func (b *Builder) resolveGithubUrlComponents(url string) (string, string, error) {
```

Comment:
Not sure about the name of this function. Maybe smth more specific like `extractOrgAndRepo` or at least just add comment which components does this method extract.

Reply:
Doing both :)